src/server.c

```diff
@@ -5183,9 +5219,16 @@ void rejectCommandFormat(client *c, const char *fmt, ...) {
 /* This is called after a command in call, we can do some maintenance job in it. */
 void afterCommand(client *c) {
     UNUSED(c);
-    /* Flush pending invalidation messages only when we are not in nested call.
-     * So the messages are not interleaved with transaction response. */
-    if (!server.in_nested_call) trackingHandlePendingKeyInvalidations();
+    if (!server.in_nested_call) {
+        /* If we are at the top-most call() we can propagate what we accumulated.
+         * Should be done before trackingHandlePendingKeyInvalidations so that we
+         * reply to client before invalidating cache (makes more sense) */
+        if (server.core_propagates)
+            propagatePendingCommands();
+        /* Flush pending invalidation messages only when we are not in nested call.
+         * So the messages are not interleaved with transaction response. */
+        trackingHandlePendingKeyInvalidations();
```

Comment:
i just realized that this call in `afterCommand` is gonna propagate the evicted DELs too, right?
which will add them to a MULTI with the command itself, or on their own if the command was a read-only command.
i think that's wrong, and that we should flush the queue after eviction.
maybe, we need to skip that though, if we're in a nested call (which is only `RM_Call, right?).
please add tests (if possible for both cases).

Reply:
afterCommand isn't called after eviction
We flush the propagation queue in performEvictions
Which is called either from the timer or process Command
I even assert that the queue is empty before eviction/expiry
so that before eviction/expiry the queue is empty, we DELete keys, and we flush the queue so now it's empty again

or maybe I misunderstood your comment

Reply:
sorry, for some reason i looked at the code and it seemed as if you didn't modify evict.c (some glitch)